### PR TITLE
Implement video-seconds-played analytics trigger

### PR DESCRIPTION
**Pull request by cvializ**
Implements the last P0 for #9902

### CODE REVIEW

File: extensions/amp-analytics/0.1/events.js [hunk 1]
```diff
@@ -470,6 +474,19 @@ export class VideoEventTracker extends EventTracker {
...
       }
 
+      if (normalizedType === VideoAnalyticsEvents.SECONDS_PLAYED) {
+        if (interval) {
+          intervalCounter++;
+          if (intervalCounter % interval !== 0) {
+            return;
+          }
+        } else {
+          user().error(TAG, 'video-seconds-played requires interval spec ' +
```
Comment (aghassemi) [inline extensions/amp-analytics/0.1/events.js]: possible to move the validity check up maybe at the time `videoSpec['interval']` is read? I assume `eventType` can be used to detect whether interval is required or not.

File: src/service/video-manager-impl.js [hunk 2]
```diff
@@ -118,6 +124,38 @@ export class VideoManager {
...
     /** @private {?VideoEntry} */
     this.dockedVideo_ = null;
+
+    /** @private {?./timer-impl.Timer} */
+    this.timer_ = null;
+
+    /** @private {?Function} */
+    this.boundSecondsPlaying_ = null;
+
+    const isAnalyticsEnabled = getMode().test ?
```
Comment (aghassemi) [inline src/service/video-manager-impl.js]: I guess if we can't detect whether they have video-analytics or not, this optimization won't really help much ( I expect vast majority of AMP pages have some sort of analytics anyway). let's replace with a TODO + GitHub issue to expose analytics config to components somehow to enable optimizations we want.

File: src/video-interface.js [hunk 3]
```diff
@@ -351,6 +351,16 @@ export const VideoAnalyticsEvents = {
    * @event video-session-visible
    */
   SESSION_VISIBLE: 'video-session-visible',
+
+  /**
+   * video-seconds-played
+   *
+   * Indicates that a video was playing when the was playing when the
```
Comment (aghassemi) [inline src/video-interface.js]: broken sentence.

File: test/integration/test-video-players-helper.js [hunk 4]
```diff
@@ -274,6 +274,41 @@ export function runVideoPlayerIntegrationTests(
...
+          timer.promise(2000),
+        ]);
+      }).then(() => {
+        const viewport = video.implementation_.getViewport();
+        viewport.scrollIntoView(video);
+        return listenOncePromise(video, VideoAnalyticsEvents.SECONDS_PLAYED);
+      }).then(() => {
+        pauseButton.click();
+        return listenOncePromise(video, VideoEvents.PAUSE);
+      }).then(() => {
```
Comment (aghassemi) [inline test/integration/test-video-players-helper.js]: let's add another `then` and play the video again to ensure seconds_played comes back

Comment (cvializ) [inline test/integration/test-video-players-helper.js]: Done.

**Pull request merged**